MoreLinq.Test/ReturnTest.cs

```diff
@@ -0,0 +1,145 @@
+            Assert.That(MoreEnumerable.Return(item), Has.Member(item));
+        }
+
+        [Test]
+        public void TestResultingListHasCountOne()
+        {
+            Assert.That(((IList<object>) MoreEnumerable.Return(new object())).Count, Is.EqualTo(1));
+        }
+
+        [Test]
+        public void TestContainsReturnsTrueWhenTheResultingSequenceContainsTheItemProvided()
+        {
+            var item = new object();
+            Assert.That(MoreEnumerable.Return(item).Contains(item), Is.True);
+        }
+
+        [Test]
+        public void TestContainsDoesNotThrowWhenTheItemContainedIsNull()
+        {
+            Assert.That(() => MoreEnumerable.Return(new object()).Contains(null), Throws.Nothing);
+        }
+
+        [Test]
+        public void TestContainsDoesNotThrowWhenTheItemProvidedIsNull()
+        {
+            Assert.That(() => MoreEnumerable.Return<object>(null).Contains(new object()), Throws.Nothing);
+        }
+
+        [Test]
+        public void TestIndexOfDoesNotThrowWhenTheItemProvidedIsNull()
+        {
+            Assert.That(() => ((IList<object>) MoreEnumerable.Return<object>(null)).IndexOf(new object()),
+                Throws.Nothing);
+        }
+
+        [Test]
+        public void TestIndexOfDoesNotThrowWhenTheItemContainedIsNull()
+        {
+            Assert.That(() => ((IList<object>)MoreEnumerable.Return<object>(new object())).IndexOf(null),
+                Throws.Nothing);
+        }
+
+        [Test]
+        public void TestCopyToSetsTheValueAtTheIndexToTheItemContained()
+        {
+            var array = new object[1];
+            var item = new object();
+
+            ((IList<object>)MoreEnumerable.Return(item)).CopyTo(array, 0);
+
+            Assert.That(array[0], Is.EqualTo(item));
+        }
+
+        [Test]
+        public void TestResultingCollectionIsReadOnly()
+        {
+            Assert.That(((ICollection<object>)MoreEnumerable.Return(new object())).IsReadOnly, Is.True);
+        }
+
+        [Test]
+        public void TestResultingCollectionHasCountOne()
+        {
+            Assert.That(((ICollection<object>)MoreEnumerable.Return(new object())).Count, Is.EqualTo(1));
+        }
+
+        [Test]
+        public void TestIndexZeroContainsTheItemProvided()
+        {
+            var item = new object();
+            Assert.That(((IList<object>) MoreEnumerable.Return(item))[0], Is.EqualTo(item));
+        }
+
+        [Test]
+        public void TestIndexOfTheItemProvidedIsZero()
+        {
+            var item = new object();
+            Assert.That(((IList<object>)MoreEnumerable.Return(item)).IndexOf(item), Is.EqualTo(0));
+        }
+
+        [Test]
+        public void TestIndexOfAnItemNotContainedIsNegativeOne()
+        {
+            Assert.That(((IList<object>)MoreEnumerable.Return(new object())).IndexOf(new object()), Is.EqualTo(-1));
+        }
+
+        private static readonly IEnumerable<Action> UnsupportedActions =
```

Comment:
Also, this would be better as `ITestCaseData` objects so you can set the test case name. Otherwise, this is what my runner shows:

```
    TestUnsupportedMethodsShouldThrow (6 tests) [0:00.006] Success
     TestUnsupportedMethodsShouldThrow(System.Action) [0:00.001] Success
     TestUnsupportedMethodsShouldThrow(System.Action) [2] [0:00.001] Success
     TestUnsupportedMethodsShouldThrow(System.Action) [3] [0:00.001] Success
     TestUnsupportedMethodsShouldThrow(System.Action) [4] [0:00.001] Success
     TestUnsupportedMethodsShouldThrow(System.Action) [5] [0:00.001] Success
     TestUnsupportedMethodsShouldThrow(System.Action) [6] [0:00.001] Success
```

Reply:
This still shows up a little wonky in the VS test explorer due to [this bug](https://github.com/nunit/nunit3-vs-adapter/issues/607).